### ISSUE TITLE
Template parameter causes fatal error on Special:Ask

### ISSUE THREAD

**Issue by JeroenDeDauw** [JeroenDeDauw/Maps#564]
https://sandbox.semantic-mediawiki.org/w/index.php?title=Sp%C3%A9cial:Ask&x=-5B-5BHas-20coordinates%3A%3A%2B-5D-5D%2F-3FHas-20coordinates&format=leaflet&template=boom

Works fine when removing template parameter.

**Comment by JeroenDeDauw** [comment]
@mwjames any idea how to fix this? I tried a bunch of things and always ended up with the parser misbehaving in some way.

**Comment by JeroenDeDauw** [comment]
* [Parser invocation](https://github.com/JeroenDeDauw/Maps/blob/master/src/SemanticMW/ResultPrinters/QueryHandler.php#L455-L457)
* [Used parser](https://github.com/JeroenDeDauw/Maps/blob/master/src/SemanticMW/ResultPrinters/QueryHandler.php#L479-L481)

**Comment by mwjames** [comment]
This is not really a `Special:Ask` issue, for a more elaborate
exposition of the issue caused by an incomplete Parser instance, see
[0].

The `RecursiveTextProcessor` avoids direct access to `wgParser` by
ensuring a Title instance is set and avoids the issue when other
extensions don't check for NULL.

[0] https://github.com/SemanticMediaWiki/semantic-mediawiki.org/issues/63

**Comment by JeroenDeDauw** [comment]
`RecursiveTextProcessor` has  @private tag on it, which I figure means it is package private. So how does that help extensions adding a result format? I'm also seeing this issue in ModernTimeline.

**Comment by mwjames** [comment]
> `RecursiveTextProcessor` has  @private tag on it, which I figure means it is
> package private. So how does that help extensions adding a result format?
> I'm also seeing this issue in ModernTimeline.

Yes, for a result printer that extends from `ResultPrinter` it will
have access to the instance for others just look at what it does and
copy the relevant Title code path.
